clorm/orm/core.py

```diff
@@ -1196,20 +1195,32 @@ class DateField(StringField):
 
     """
     def __init__(self, default: Any=MISSING, index: Any=MISSING) -> None:
-        self._default = (True, default) if default is not MISSING else (False, None)
         self._index = index if index is not MISSING else False
 
         if default is MISSING:
+            self._default = (False, None)
             return
 
-        # Check that the default is a valid value. If the default is a callable then
-        # we can't do this check because it could break a counter type procedure.
-        if not callable(default):
-            try:
+        self._default = (True, default)
+        cmplx = self.complex
+
+        # Check and convert the default to a valid value. Note: if the default
+        # is a callable then we can't do this check because it could break a
+        # counter type procedure.
+        if callable(default) or (cmplx and isinstance(default, cmplx)):
+            return
+
+        try:
+            if cmplx:
+                if cmplx.meta.is_tuple:
+                    self._default = (True, _instance_from_tuple(cmplx, default))
+                else:
+                    raise default
```

Comment:
It was sort of intentional, but I was being lazy :). The main thing I wanted was to raise either a TypeError or ValueError so that the surrounding try-except block would catch the exception and then re-throw a TypeError("Invalid default value ...") exception. But I'll change it to explicitly raise a ValueError.